### PR TITLE
Delete Non-current versions

### PR DESCRIPTION
**Pull request by bexsoft**
## What does this do?

- Delete Non-current API
- Delete non current modal implementation

## How does it look?

Screen Shot 2022-03-17 at 22 41 15
Screen Shot 2022-03-17 at 22 41 02
Screen Shot 2022-03-17 at 22 40 57

Signed-off-by: Benjamin Perez <benjamin@bexsoft.net>

### CODE REVIEW

File: portal-ui/src/icons/DeleteNonCurrentIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2021 MinIO, Inc.
```
Comment (dvaldivia) [inline portal-ui/src/icons/DeleteNonCurrentIcon.tsx:2]:
```suggestion
// Copyright (c) 2022 MinIO, Inc.
```

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/DeleteNonCurrent.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2021 MinIO, Inc.
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/DeleteNonCurrent.tsx:2]:
```suggestion
// Copyright (c) 2022 MinIO, Inc.
```

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by dvaldivia — DISMISSED** [review]

**Comment by jinapurapu** [comment]
Should the icon for "Delete all Versions" be hidden if only one version remains?

**Review by adfost — APPROVED** [review]

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ObjectDetails/VersionsNavigator.tsx [hunk 3]
```diff
@@ -417,6 +439,17 @@ const VersionsNavigator = ({
                 }
                 actions={
                   <Fragment>
+                    <RBIconButton
```
Comment (jinapurapu) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ObjectDetails/VersionsNavigator.tsx:442]:
```suggestion
{filteredRecords.length > 1 && ( <RBIconButton
                 ```

**Review by jinapurapu — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ObjectDetails/VersionsNavigator.tsx (hunk 3 above)
Comment (bexsoft) [reply portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ObjectDetails/VersionsNavigator.tsx:442]: Fixed by disabling the button:
Screen Shot 2022-03-18 at 13 58 23
Screen Shot 2022-03-18 at 13 58 10

**Review by dvaldivia — APPROVED** [review]

**Review by jinapurapu — APPROVED** [review]

**Pull request merged**